sdk/storage/azblob/appendblob/client.go

```diff
@@ -200,6 +200,12 @@ func (ab *Client) SetTier(ctx context.Context, tier blob.AccessTier, o *blob.Set
 	return ab.BlobClient().SetTier(ctx, tier, o)
 }
 
+// SetExpiry operation sets an expiry time on an existing blob. This operation is only allowed on Hierarchical Namespace enabled accounts.
+// For more information, see https://learn.microsoft.com/en-us/rest/api/storageservices/set-blob-expiry
+func (ab *Client) SetExpiry(ctx context.Context, expiryType blob.ExpiryType, o *blob.SetExpiryOptions) (blob.SetExpiryResponse, error) {
```

Comment:
Does the page blob client need this too?  I don't see it being added.

Reply:
Set expiry operation is only allowed on HNS enabled accounts.
HNS enabled accounts does not support page blobs. This is why I haven't added this to the page blob client.